common/src/main/java/com/linecorp/centraldogma/common/AggregationQuery.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * An aggregation query on files.
+ *
+ * @param <T> the content type of files being queried
+ */
+public interface AggregationQuery<T> {
+
+    /**
+     * Returns a newly-created {@link AggregationQuery} that aggregates the JSON contents in the specified
+     * {@code pathAndOptionals}. Then, the specified
+     * <a href="https://github.com/json-path/JsonPath/blob/master/README.md">JSON path expressions</a>
+     * are applied to the content.
+     *
+     * @param pathAndOptionals the paths of JSON files being aggregated and whether it is optional
+     * @param jsonPaths the JSON path expressions to apply
+     */
+    static AggregationQuery<JsonNode> ofAggregationJson(Iterable<PathAndOptional> pathAndOptionals,
```

Comment:
I think we need to be consistent with `Query` API:

- `ofJson()` - identity aggregation query (which could be just a shortcut of `ofJsonPath()` with empty expressions unlike `Query` though.)
- `ofJsonPath()` - JSON aggregation query